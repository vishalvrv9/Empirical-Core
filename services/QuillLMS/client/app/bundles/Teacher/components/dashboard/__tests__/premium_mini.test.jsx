import React from 'react';
import { shallow } from 'enzyme';
import $ from 'jquery';

import PremiumMini from '../premium_mini';

jest.mock('jquery', () => ({
  post: jest.fn().mockReturnValue({
    success: jest.fn(),
  }),
}));

describe('PremiumMini component', () => {
  it('should render', () => {
    expect(shallow(<PremiumMini />)).toMatchSnapshot();
  });

  it('should post to subscriptions on button click', () => {
    const wrapper = shallow(<PremiumMini userIsEligibleForTrial />);
    wrapper.find('button').simulate('click');
    expect($.post.mock.calls[0][0]).toBe('/subscriptions');
    expect($.post.mock.calls[0][1].subscription.account_type).toBe('Teacher Trial');
  });
});
