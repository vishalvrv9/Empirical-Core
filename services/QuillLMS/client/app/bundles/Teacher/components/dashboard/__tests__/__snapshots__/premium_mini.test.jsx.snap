// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PremiumMini component should render 1`] = `
<div
  className="mini_container results-overview-mini-container col-md-4 col-sm-5 text-center"
>
  <div
    className="mini_content"
  >
    <div
      className="premium-container "
    >
      <h4>
        Try Premium for Free
      </h4>
      <button
        className="btn btn-orange"
        onClick={[Function]}
        type="button"
      >
        Get Premium Free for 30 days
      </button>
      <p
        className="credit-card"
      >
        No credit card required.
      </p>
      <p>
        Unlock your Premium trial to save time grading and gain actionable insights.
      </p>
      <a
        href="https://support.quill.org/quill-premium"
        target="_blank"
      >
        Learn more about Premium
      </a>
    </div>
  </div>
</div>
`;
